packages/ckeditor5-show-blocks/theme/showblocks.css

```diff
@@ -2,3 +2,107 @@
  * Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
  * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
  */
+
+@import "@ckeditor/ckeditor5-ui/theme/mixins/_dir.css";
+
+:root {
+	--ck-show-blocks-border-color: hsl(0, 0%, 46%);
+}
+
+@define-mixin block-name-background $text {
+	background-image: url("data:image/svg+xml;utf8,<svg width='120' height='12' xmlns='http://www.w3.org/2000/svg' ><text style='paint-order:stroke fill; clip-path: inset(-3px);' stroke='%23EAEAEA' stroke-width='13' dominant-baseline='middle' fill='black' x='3' y='7' font-size='9px' font-family='Consolas, %22Lucida Console%22, %22Lucida Sans Typewriter%22, %22DejaVu Sans Mono%22, %22Bitstream Vera Sans Mono%22, %22Liberation Mono%22, Monaco, %22Courier New%22, Courier, monospace'>$(text)</text></svg>");
+	background-repeat: no-repeat;
+	background-position: 1px 1px;
+	padding-top: 15px;
+
+	&:not(.ck-widget_selected):not(.ck-widget:hover) {
+		outline: 1px dashed var(--ck-show-blocks-border-color);
+	}
+
+	@mixin ck-dir rtl {
+		background-image: url("data:image/svg+xml;utf8,<svg width='120' height='12' xmlns='http://www.w3.org/2000/svg' ><text style='paint-order:stroke fill; clip-path: inset(-3px); transform:translate(-2px, 0);' stroke='%23EAEAEA' stroke-width='13' dominant-baseline='middle' fill='black' x='100%' text-anchor='end' y='7' font-size='9px' font-family='Consolas, %22Lucida Console%22, %22Lucida Sans Typewriter%22, %22DejaVu Sans Mono%22, %22Bitstream Vera Sans Mono%22, %22Liberation Mono%22, Monaco, %22Courier New%22, Courier, monospace'>$(text)</text></svg>");
+		background-position: calc(100% - 1px) 1px;
+	}
+}
+
+.ck.ck-editor__editable.ck-show-blocks {
+	& address {
+		@mixin block-name-background ADDRESS;
+	}
+
+	& aside {
+		@mixin block-name-background ASIDE;
+	}
+
+	& blockquote {
+		@mixin block-name-background BLOCKQUOTE;
+	}
+
+	& > div {
```

Comment:
I'm not so sure about this. If this was meant to exclude widget internals, this is too wide. For instance, this will hide divs inside some GHS structures (`article > div`) the user would certainly want to notice.